src/esphome/sensor/mpr121_sensor.cpp

```diff
@@ -0,0 +1,90 @@
+#include "esphome/defines.h"
+#ifdef USE_MPR121
+
+#include "esphome/sensor/mpr121_sensor.h"
+#include "esphome/log.h"
+#include <bitset>
+
+ESPHOME_NAMESPACE_BEGIN
+
+namespace sensor {
+
+static const char *TAG = "sensor.mpr121";
+
+MPR121SensorChannel::MPR121SensorChannel(uint8_t channel, float value) {
+  this->channel = channel;
+  this->value = value;
+}
+
+MPR121Sensor::MPR121Sensor(const std::string &name) : Sensor(name) {
+  this->state = 0.0;
+ }
+
+void MPR121Sensor::process(uint16_t *data, uint16_t *last_data) {
+  //process depend on the sensor_type_
+  switch(this->sensor_type_) {
+    case MPR121_SENSOR_TYPE_SLIDER:
+      this->process_slider_(data);
+      break;
+    case MPR121_SENSOR_TYPE_ROTARY:
+      this->process_rotary_(data, last_data);
+      break;
+    default:
+      ESP_LOGE(TAG, "'%s': Unknown sensor type %d", this->name_.c_str(), this->sensor_type_);
+  }
+}
+
+void MPR121Sensor::process_slider_(uint16_t *data) {
+  uint8_t count = 0;
+  float value = 0;
+  for(auto *chan : this->channels_) {
+    if (*data & (1 << chan->channel)) {
+      value += chan->value;
+      count++;
+    }
+  }
+  if(count > 0) {
+    this->publish_state(value/count);
+    ESP_LOGD(TAG, "'%s': Sending new state %f", this->name_.c_str(), state);
+  }
+}
+
+void MPR121Sensor::process_rotary_(uint16_t *data, uint16_t *last_data) {
+  std::bitset<16> bits = 0;
+  std::bitset<16> last_bits = 0;
+  float value = 0;
+  // get the bit value for all channels
+  bits |= (*data & this->mask_);
+  last_bits |= (*last_data & this->mask_);
+
+  ESP_LOGD(TAG, "'%s': bits %s", this->name_.c_str(), bits.to_string().c_str());
+  if(last_bits._Find_first() <= bits._Find_first()) {
```

Comment:
`_Find_first` appears to be an internal method (at least not documented in std). We shouldn't depend on that, please implement an own method for that.